pkg/controller/identity/iamaccesskey/controller.go

```
@@ -0,0 +1,175 @@
+				}, nil
+			}
+		}
+	}
+
+	return managed.ExternalObservation{}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMAccessKey)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	response, err := e.client.CreateAccessKeyRequest(&awsiam.CreateAccessKeyInput{UserName: aws.String(cr.Spec.ForProvider.IAMUsername)}).Send(ctx)
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+
+	var conn managed.ConnectionDetails = nil
+	if response != nil && response.AccessKey != nil {
+		conn = managed.ConnectionDetails{}
+		conn[runtimev1alpha1.ResourceCredentialsSecretUserKey] = []byte(aws.StringValue(response.AccessKey.AccessKeyId))
+		conn[runtimev1alpha1.ResourceCredentialsSecretPasswordKey] = []byte(aws.StringValue(response.AccessKey.SecretAccessKey))
+
+		cr.Spec.ForProvider.Status = string(response.AccessKey.Status)
+
+		meta.SetExternalName(cr, aws.StringValue(response.AccessKey.AccessKeyId))
+
+		err := e.kube.Update(ctx, cr)
+		if err != nil {
+			return managed.ExternalCreation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	return managed.ExternalCreation{ConnectionDetails: conn}, errors.Wrap(err, errCreate)
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.IAMAccessKey)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	_, err := e.client.UpdateAccessKeyRequest(&awsiam.UpdateAccessKeyInput{
+		AccessKeyId: aws.String(meta.GetExternalName(cr)),
+		Status:      awsiam.StatusType(cr.Spec.ForProvider.Status),
+		UserName:    aws.String(cr.Spec.ForProvider.IAMUsername),
+	}).Send(ctx)
+
+	return managed.ExternalUpdate{}, errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errUpdate)
```

```suggestion
	return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
```

If the resource does not exist, we should return an error here.